chainerx_cc/cmake/FindBLAS.cmake

```diff
@@ -171,6 +171,21 @@ macro(Check_Fortran_Libraries LIBRARIES _prefix _name _flags _list _thread)
     mark_as_advanced(${_prefix}${_combined_name}_WORKS)
     set(_libraries_work ${${_prefix}${_combined_name}_WORKS})
   endif()
+  if(_libraries_work)
+    # NOTE: ChainerX uses CBLAS API. Search `libcblas.so` if CBLAS functions are
+    # not provided in the blas library found.
+    set(CMAKE_REQUIRED_LIBRARIES ${_flags} ${${LIBRARIES}} ${_thread})
+    check_function_exists("cblas_sgemm" ${_prefix}${_combined_name}_HAS_CBLAS)
+    if(NOT ${_prefix}${_combined_name}_HAS_CBLAS)
+        find_library(${_prefix}${_combined_name}_CBLAS_LIBRARY cblas)
+        if(${_prefix}${_combined_name}_CBLAS_LIBRARY)
+          set(${LIBRARIES} ${${LIBRARIES}} ${${_prefix}${_combined_name}_CBLAS_LIBRARY})
+        else()
+          set(_libraries_work FALSE)
```

Comment:
Could you explain why this is needed?

Reply:
* If `cblas_sgemm` symbol not found in the blas library (e.g. `libopenblas.so`), then `${_prefix}${_combined_name}_HAS_CBLAS` is set false.
* In this case, check whether `libcblas.so` exists. This file should provides CBLAS API.
* If `libcblas.so` not found, then we couldn't use CBLAS API (only FORTRAN API available.)  So we revert `_li
brafy_work` flag.

This situation may happen when user is running Arch Linux. It strips CBLAS API from OpenBLAS, and provides the API by a separate package [`cblas`](https://www.archlinux.org/packages/extra/x86_64/cblas/) that contains `libcblas.so`.

Reply:
Thanks for the explanation. I just want to make sure but the vendor for which this path is passed is OpenBLAS? Is it the case that no other vendors are available (e.g. ATLAS) in the environment for which this issue tries to address? If not, you could use `BLA_VENDOR ` to specify the candidate vendors explicitly. For reference, this seems like a step towards a less ad-hoc approach to me, but this also depends on the answers to those questions. https://cmake-basis.github.io/apidoc/latest/FindBLAS_8cmake_source.html

Reply:
ATLAS has the same issue. ATLAS officially provides `libatlas.so` and `libcblas.so` separately.

https://packages.ubuntu.com/ja/bionic/amd64/libatlas3-base/filelist

Debian and Ubuntu also build `libblas.so` to support BLAS library switching. Since we blacklisted `libatlas.so`, the issue hadn't appear on those distributions.

https://salsa.debian.org/science-team/atlas/blob/a53f04c697a75dd39487fe331d49eea13effefa7/debian/rules#L178-182

Reply:
I see. I just find the current change a bit ad-hoc since it's non-trivial and not visible outside `Check_Fortran_Libraries`. How about adding another argument to it, for additional libraries to search for (and changing the `_name` passed to the function for OpenBLAS and ATLAS to the "missing" symbol?